runtime/libia2/ia2.c

```diff
@@ -246,6 +246,8 @@ static int segment_flags_to_access_flags(Elf64_Word flags) {
       ((flags & PF_R) != 0 ? PROT_READ : 0);
 }
 
+uintptr_t tls_addr[16][2] IA2_SHARED_DATA = {0};
```

Comment:
Can we use add `const int`s for these integer literals? I understand the first is number of pkeys, but I don't know what the other one is.

Reply:
Yeah.  I was gonna ask if there's a macro for the max number of compartments.  The 2 is just because sometimes the TLS regions are on the side (so there are two) and sometimes it's just whole.

Reply:
I don't understand why TLS is split if we don't support shared TLS yet, but I'll take a look. Either way shouldn't it show up as a single segment in proc maps?

Reply:
Only the first compartment should have a split TLS region so I think this can be a 16 entry array plus 2 more variables for the split TLS.

Reply:
Yeah, (reiterating from Slack for posterity) the split TLS region for compartment 0 is because there is a page of unprotected data for `ia2_stackptr_0` plus padding, as we don't have a general implementation of shared TLS yet but `ia2_stackptr_0` is special-cased for now as it must be stored in TLS and unprotected.